asset/asset.go

```diff
@@ -1038,6 +1084,17 @@ func (a *Asset) Encode(w io.Writer) error {
 	return stream.Encode(w)
 }
 
+// EncodeNoWitness encodes the asset without the witness into a TLV stream.
+// This is used for serializing on an asset as a leaf within a TAP MS-SMT tree.
+// This only applies when the asset version is v1.
+func (a *Asset) EncodeNoWitness(w io.Writer) error {
+	stream, err := tlv.NewStream(a.encodeRecords(EncodeSegwit)...)
```

Comment:
How about switching on the asset version in `Asset.Encode(io.Writer)` and then dropping `EncodeNoWitness`? We can also then get rid of doc comments like `This only applies when the asset version is v1.`.

Reply:
So we don't want to unilaterally change the asset encoding. As an example, we want to ensure the asset serialized in proofs still has the witness. The goal here is only to change the serialization when we insert into the SMT tree which is committed to in the top level taproot output.

Reply:
> we want to ensure the asset serialized in proofs still has the witness

Even `V1` assets serialized in proofs still has the witness, is what I think you're saying. :+1: